### PR TITLE
More gen 22: Flow coverage exploration (remade)

### PR DESCRIPTION
**Pull request by chrisnojima**
Replaced https://github.com/keybase/client/pull/9575 . See that for comments

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/actions/chat/inbox.js [hunk 1]
```diff
@@ -423,7 +424,7 @@ function* _chatInboxFailedSubSaga(params) {
       break
     }
     case RPCChatTypes.localConversationErrorType.otherrekeyneeded: {
-      const rekeyers = error.rekeyInfo.rekeyers
+      const rekeyers = (error.rekeyInfo && error.rekeyInfo.rekeyers) || []
       yield Saga.put(ChatGen.createUpdateInboxRekeyOthers({conversationIDKey, rekeyers}))
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/actions/chat/inbox.js]: unclear, i don't want things to really logically change so lets keep it for now

File: shared/chat/inbox/container.js [hunk 2]
```diff
@@ -202,15 +220,18 @@ const mapStateToProps = (state: TypedState, {isActiveRoute, routeState}) => {
...
 }
 
-const mapDispatchToProps = (dispatch: Dispatch, {focusFilter, routeState, setRouteState}) => ({
+const mapDispatchToProps = (dispatch: any => void, {focusFilter, routeState, setRouteState}: OwnProps) => ({
   loadInbox: () => dispatch(ChatGen.createLoadInbox()),
-  _onSelectNext: (rows, direction) => dispatch(ChatGen.createSelectNext({rows, direction})),
-  onHotkey: cmd => {
+  _onSelectNext: (rows: Array<Inbox.RowItemSmall | Inbox.RowItemBig>, direction: -1 | 1) =>
+    dispatch(
+      ChatGen.createSelectNext({rows: rows.map(r => ({conversationIDKey: r.conversationIDKey})), direction})
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/chat/inbox/container.js]: i think we should only send what we need and likely the answer is to make that type exact

File: shared/actions/chat/inbox.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/actions/chat/inbox.js]: ok, sounds good

**Pull request merged**